app/services/user_service.rb

```diff
@@ -10,13 +10,15 @@ class UserService
     attr_accessor :user_data_source
 
     def initialize(user_data_source: nil)
-      unless user_data_source.nil?
-        raise "user_data_source must be descendant of #{UserDataSource.name}" unless user_data_source.class <= UserDataSource
+      if !user_data_source.nil? && user_data_source.class > UserDataSource
```

Comment:
so much more confusing using `if !` instead of `unless`

Reply:
Yeah? I find having multil-expression conditionals with unless statements sometimes difficult to reason about. I think Rubocop suggests that you use a negated `if` when you have `&&` or `||`, but I'd have to check to be sure.